include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -565,6 +573,50 @@ def ObserveAnsatz : Pass<"observe-ansatz", "mlir::func::FuncOp"> {
   ];
 }
 
+def StatePreparation : Pass<"state-prep", "mlir::ModuleOp"> {
```

Comment:
Why is StateDecomposer not listed here?

Reply:
It looks like it is a part of StatePrep and not its own pass.

Reply:
It is not a pass - it is helper code that converts vectors to gates using a builder

Reply:
I moved the code from StateDecomposer.h and StateDecomposer.cpp into StatePreparation.cpp